pkg/virt-controller/watch/migration/migration.go

```diff
@@ -393,7 +407,15 @@ func (c *Controller) canMigrateVMI(migration *virtv1.VirtualMachineInstanceMigra
 	}
 
 	return true, nil
+}
 
+func (c *Controller) failMigration(migration *virtv1.VirtualMachineInstanceMigration) error {
+	err := backendstorage.MigrationAbort(c.clientset, migration)
+	if err != nil {
+		return err
+	}
+	migration.Status.Phase = virtv1.MigrationFailed
```

Comment:
I disagree. Just like a migration cannot succeed before backend-storage handoff, it cannot fail before the target PVC has been removed. Otherwise, that PVC may never get removed.

Reply:
I don't like the fact that the migration may stay pending forever but I think you're right.

Anyway if there's an infra issue (causing the pvc to be deleted) it shouldn't be KubeVirt trying mitigating it.